doc/rst/technotes/initializers.rst

```diff
@@ -81,36 +81,65 @@ The first initializer lets the user specify the initial coordinates and the
 second initializer lets the user specify the initial message when creating a
 MessagePoint.
 
+Order of Initialization
+-----------------------
+
+The specification of a class may be derived from, or inherit from, the
+specification of one or more other classes.  The initialization of an
+instance of a derived class requires that the initializer for each parent
+class be executed in some well defined order.
+
+Chapel initializes an instance in two phases that we refer to as "Phase 1"
+and "Phase 2".
+
+Phase 1 proceeds from the most derived class to the base class, and the fields
+for each class are initialized in field declaration order. This implies that
+the fields of any parent classes will be in an undefined state during phase 1.
+This in turn requires that certain constraints be observed during phase 1.
+These are described in more detail below.
+
+Phase 2 proceeds from the base class to the most derived class once phase 1
+has been completed for the base class.  At this point every field of the
+instance is in a well defined state and so there are no restrictions on
+the operations that may be performed.
```

Comment:
This vaguely feels a little too much like a rationale, but I might be being overly sensitive to that